engine/api/src/main/scala/pl/touk/nussknacker/engine/api/typed/typing.scala

```diff
@@ -160,6 +160,18 @@ object typing {
       }
     }
 
+    def apply(obj: TypedMap): TypingResult = {
```

Comment:
maybe we could have apply(obj: AnyRef): TypingResult? 
In other cases (currently) it would just do Typed(obj.getClass)

Reply:
I thought about it, but have some doubts that this approach will make `Typed` api to be unclear, because `Class | ClassRef <: AnyRef`. I know that we have priorities in method overloading but still I think that it is not the best api design choice. Maybe we can name this factory method? It should be done carefully because we already have: `apply`, `detailed`. Maybe we should name this `fromInstance`? Than for consistence `detailed` will be renamed to `fromDetailedType` or sth similar. WDYT?

Reply:
yes, can be some other method, or even other class - but let's keep it in one place

Reply:
Ok, added to Typed as a named factory method. Also renamed detailed -> fromDetailedType for consistency.